src/hostapi/wasapi/pa_win_wasapi.c

```diff
@@ -3910,7 +3951,7 @@ static PaError OpenStream( struct PaUtilHostApiRepresentation *hostApi,
         if (stream->in.params.blocking == TRUE)
         {
             UINT32 bufferFrames = ALIGN_NEXT_POW2((stream->in.framesPerHostCallback / WASAPI_PACKETS_PER_INPUT_BUFFER) * 2);
-            UINT32 frameSize    = stream->in.wavexu.ext.Format.nBlockAlign;
+            UINT32 frameSize    = stream->in.wavexu.ext.Format.nBlockAlign / (stream->in.monoMixer != NULL ? 2 : 1);
```

Comment:
```suggestion
            UINT32 frameSize    = (stream->in.monoMixer != NULL) ? (stream->in.wavexu.ext.Format.nBlockAlign / 2) : stream->in.wavexu.ext.Format.nBlockAlign;
```
More likely to get strength reduction to right shift this way.

Reply:
It depends on how smart compiler is, it could do right shift in my case too, if deducted, because the result of the comparison is constant (1 or 2). I would avoid doing this micro optimization because in my case the comparison is less bulky. Please let me know your opinion.